lib/ruby_lsp/requests/hover.rb

```diff
@@ -0,0 +1,74 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "ruby_lsp/requests/support/rails_document_client"
+
+module RubyLsp
+  module Requests
+    # ![Hover demo](../../misc/rails_document_link_hover.gif)
+    #
+    # The [hover request](https://microsoft.github.io/language-server-protocol/specification#textDocument_hover)
+    # renders a clickable link to the code's official documentation.
+    # It currently only supports Rails' documentation: when hovering over Rails DSLs/constants under certain paths,
+    # like `before_save :callback` in `models/post.rb`, it generates a link to `before_save`'s API documentation.
+    #
+    # # Example
+    #
+    # ```ruby
+    # class Post < ApplicationRecord
+    #   before_save :do_something # when hovering on before_save, the link will be rendered
+    # end
+    # ```
+    class Hover < BaseRequest
+      extend T::Sig
+
+      sig { params(document: Document, uri: String, position: Document::PositionShape).void }
+      def initialize(document, uri, position)
+        super(document)
+
+        @file_dir = T.let(Pathname.new(uri).dirname.to_s, String)
+        @hover = T.let(nil, T.nilable(LanguageServer::Protocol::Interface::Hover))
+        position = Document::Scanner.new(document.source).find_position(position)
+
+        target, _ = locate_node_and_parent(
```

Comment:
@vinistock Now it's just one tree traversal and it's happen in `#initialize`. We don't even need the visitor methods anymore.

Reply:
Actually, I decided to move this part to `#run` instead.